From diegoliz@gmail.com  Tue Apr 21 15:49:50 2009
From: Diego Liziero <diegoliz@gmail.com>
Date: Tue, 14 Apr 2009 04:33:54 +0200
Subject: Staging winbond: boolean negation and bitwise operation in wrong order
To: gregkh@suse.de
Message-ID: <49E3F612.5020606@gmail.com>



From: Diego Liziero <diegoliz@gmail.com>

The semantic patch that makes this change is:
(http://www.emn.fr/x-info/coccinelle/)

@@ expression E; constant C; @@
(
- !E == C
+ E != C
)

Signed-off-by: Diego Liziero <diegoliz@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/winbond/mds.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/winbond/mds.c
+++ b/drivers/staging/winbond/mds.c
@@ -432,7 +432,7 @@ Mds_Tx(struct wbsoft_priv * adapter)
 		return;
 
 	//Only one thread can be run here
-	if (!atomic_inc_return(&pMds->TxThreadCount) == 1)
+	if (atomic_inc_return(&pMds->TxThreadCount) != 1)
 		goto cleanup;
 
 	// Start to fill the data
